tough-kms/src/lib.rs

```diff
@@ -188,7 +196,130 @@ impl Sign for KmsRsaKey {
                 profile: self.profile.clone(),
                 key_id: self.key_id.clone(),
             })?;
-        let signature = response.signature.context(error::SignatureNotFound)?;
-        Ok(signature.to_vec())
+        let signature = response
+            .signature
+            .context(error::SignatureNotFound)?
+            .to_vec();
+
+        // sometimes KMS produces a signature that is shorter than the modulus. in those cases,
+        // we have observed that openssl and KMS will both validate the signature, but ring will
+        // not. if we pad the beginning of the signature with zeros to make the signature exactly
+        // the same length as the modulus, then ring will verify the signature.
+        let signature = pad_signature(signature, self.modulus_size_bytes)?;
+        Ok(signature)
+    }
+}
+
+/// Parses the `CustomerMasterKeySpec` string returned by KMS, e.g. `RSA_3072` and returns the size
+/// of the modulus in bytes. For example `RSA_3072` has a modulus of 3072 bits, so the function will
+/// return 384 == (3072 / 8). If the parsed number is not divisible by 8, an error is returned.
+fn parse_modulus_length_bytes(spec: &str) -> error::Result<usize> {
+    // only RSA is currently supported
+    ensure!(
+        spec.starts_with("RSA_"),
```

Comment:
Are we not sure whether the other types would be OK with the same padding strategy?  I guess it doesn't matter since we only support one algorithm and RSA keys?

(I was initially thinking this would make it a breaking release, but it has to be a breaking release anyway due to the change to `KmsRsaKey`...)

Reply:
The change seems specific to RSA, but I honestly don't know for sure. Elsewhere @srgothi92 recommends only running the function if the key type variant matches.

The fields of the KmsRsaKey struct are private, so the programmer interface is unchanged. It seems like it could be a patch release?

Reply:
Ah, I missed that the fields are private, no problem there then.  Do we only allow `RSA_*` because we only support `RsassaPssSha256`?  If that's true, I think there's no change in support.

Reply:
> Do we only allow `RSA_*` because we only support `RsassaPssSha256`?

Correct, we only support `RsassaPssSha256`.